app/script/user/UserRepository.js

```diff
@@ -681,13 +681,17 @@ z.user.UserRepository = class UserRepository {
 
   /**
    * Find a local user.
-   * @param {string} user_id - User ID
+   * @param {string} userId - User ID
    * @returns {Promise<z.entity.User>} Resolves with the matching user entity
    */
-  findUserById(user_id) {
-    for (const user_et of this.users()) {
-      if (user_et.id === user_id) {
-        return Promise.resolve(user_et);
+  findUserById(userId) {
+    if (!userId) {
+      return Promise.reject(new z.user.UserError(z.user.UserError.TYPE.MISSING_PARAMETER));
```

Comment:
Previously any requestor of `get_user_by_id` would have to handle undefined as a potential return value. This should never have been the case.